roadmap/implementers-guide/src/node/backing/statement-distribution.md

```diff
@@ -1,107 +1,401 @@
+
+For clarity, here is the full timeline:
+
+- candidate seconded
+- distributed along grid
+- backable in cluster
+- latecomers issue statements
+- candidate posted on chain
+- really latecomers issue statements
+
+## Cluster Module
+
+The cluster module provides direct distribution of unbacked candidates within a
+group. By utilizing this initial phase of propogating only within
+clusters/groups, we bound the number of `Seconded` messages per validator per
+relay-parent, helping us prevent spam. Validators can try to circumvent this,
+but they would only consume a few KB of memory and it is trivially slashable on
+chain.
+
+The cluster module determines whether to accept/reject messages from other
+validators in the same group. It keeps track of what we have sent to other
+validators in the group, and pending statements. For the full protocol, see
+"Protocol".
+
+## Grid Module
+
+The grid module provides distribution of backed candidates and late statements
+outside the group. For the full protocol, see the "Protocol" section.
+
+### Grid Topology
+
+For distributing outside our cluster we use a 2D grid topology. This limits the
+amount of peers we send messages to and handle view updates.
+
+The basic operation of the grid topology is that:
+
+- A validator producing a message sends it to its row-neighbors and its
+  column-neighbors.
+- A validator receiving a message originating from one of its row-neighbors
+  sends it to its column-neighbors.
+- A validator receiving a message originating from one of its column-neighbors
+  sends it to its row-neighbors.
+
+This grid approach defines 2 unique paths for every validator to reach every
+other validator in at most 2 hops, providing redundancy.
+
+Propagation follows these rules:
+
+- For groups that we are in, receive from nobody and send to our X/Y peers.
+- For groups that we are not part of:
+  - We receive from any validator in the group we share a slice with and send to
+    the corresponding X/Y slice in the other dimension.
+  - For any validators we don't share a slice with, we receive from the nodes
+    which share a slice with them.
+
+### Seconding Limit
+
+The seconding limit is a per-validator limit. Before asynchronous backing, we
+had a rule that every validator was only allowed to second one candidate per
+relay parent. With asynchronous backing, we have a 'maximum depth' which makes
+it possible to second multiple candidates per relay parent. The seconding limit
+is set to `max depth + ` to set an upper bound on candidates entering the
+system.
+
+## Candidates Module
+
+The candidates module provides a tracker for all known candidates in the view,
+and whether they are confirmed or not. What is a confirmed candidate? The first
+time a validator gets an announcement for an unknown candidate, it will send a
+request for the candidate. Upon receiving a response and validating it (see
+`UnhandledResponse::validate_response`), will mark the candidate as confirmed.
```

Comment:
A simpler case: a validator fetched collation directly from collator and validated it.
So in short: confirmed = we hold full receipt along with PVD, right?